src/main/automation/BaseActionService.ts

```
@@ -0,0 +1,383 @@
+          if (inflightRequests.size <= maxInflightRequests) {
+            resolveOnce('Page already loaded and idle');
+            return;
+          }
+        }
+
+        checkNetworkIdle();
+      } catch (error) {
+        console.error('[NetworkIdle] Setup error:', error);
+        resolveOnce('Setup error, proceeding anyway');
+      }
+    });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)